sdk/src/transaction_context.rs

```diff
@@ -57,12 +57,18 @@ impl TransactionContext {
             instruction_stack: Vec::with_capacity(instruction_context_capacity),
             number_of_instructions_at_transaction_level,
             instruction_trace: Vec::with_capacity(number_of_instructions_at_transaction_level),
-            return_data: (Pubkey::default(), Vec::new()),
+            return_data: TransactionReturnData::default(),
         }
     }
 
     /// Used by the bank in the runtime to write back the processed accounts and recorded instructions
-    pub fn deconstruct(self) -> (Vec<TransactionAccount>, Vec<Vec<InstructionContext>>) {
+    pub fn deconstruct(
+        self,
+    ) -> (
+        Vec<TransactionAccount>,
+        Vec<Vec<InstructionContext>>,
+        TransactionReturnData,
+    ) {
```

Comment:
time for a struct!

Reply:
I went with `TransactionRecord`, let me know if you prefer another name

Reply:
There's already a [`TransactionRecorder`](https://github.com/solana-labs/solana/blob/cb061263887fb20017d7b813ee9ef178a2e304f4/poh/src/poh_recorder.rs#L104) in `solana-poh`, which is pretty close :thinking: 

Maybe something like `ExecutionRecord`?

Reply:
I can get down with that -- done!